src/Symfony/Component/Mailer/Transport/Dsn.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Transport;
+
+use Symfony\Component\Mailer\Exception\InvalidArgumentException;
+
+/**
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+final class Dsn
+{
+    private $scheme;
+    private $host;
+    private $port;
+    private $user;
+    private $pass;
+    private $options;
+
+    public function __construct(string $scheme, string $host, ?int $port, string $user, string $pass, array $options = [])
+    {
+        $this->scheme = $scheme;
+        $this->host = $host;
+        $this->port = $port;
+        $this->user = $user;
+        $this->pass = $pass;
+        $this->options = $options;
+    }
+
+    public static function fromString(string $dsn): self
+    {
+        if (false === $parsedDsn = parse_url($dsn)) {
+            throw new InvalidArgumentException(sprintf('The "%s" mailer DSN is invalid.', $dsn));
+        }
+
+        if (!isset($parsedDsn['scheme'])) {
+            throw new InvalidArgumentException(sprintf('The "%s" mailer DSN must contain a transport scheme.', $dsn));
+        }
+
+        if (!isset($parsedDsn['host'])) {
+            throw new InvalidArgumentException(sprintf('The "%s" mailer DSN must contain a mailer name.', $dsn));
+        }
+
+        $user = urldecode($parsedDsn['user'] ?? '');
+        $pass = urldecode($parsedDsn['pass'] ?? '');
+        parse_str($parsedDsn['query'] ?? '', $query);
+
+        return new self($parsedDsn['scheme'], $parsedDsn['host'], $parsedDsn['port'] ?? null, $user, $pass, $query);
+    }
+
+    public function getScheme(): string
+    {
+        return $this->scheme;
+    }
+
+    public function getHost(): string
+    {
+        return $this->host;
+    }
+
+    public function getPort(int $default = null): ?int
+    {
+        return $this->port ?? $default;
+    }
+
+    public function getUser(): string
+    {
+        return $this->user;
+    }
+
+    public function getPass(): string
+    {
+        return $this->pass;
+    }
+
+    public function getOption(string $key, $default = null)
+    {
+        return \array_key_exists($key, $this->options) ? $this->options[$key] : $default;
```

Comment:
```suggestion
        return $this->options[$key] ?? $default;
```
?

Reply:
If the value is `null` and the default isn't, it should be returned imo